src/t_stream.c

```diff
@@ -1276,9 +1272,20 @@ void xaddCommand(client *c) {
     }
 
     /* Lookup the stream at key. */
-    robj *o;
     stream *s;
-    if ((o = streamTypeLookupWriteOrCreate(c,c->argv[1])) == NULL) return;
```

Comment:
we currently don't have many commands that can create a stream if it's missing, but i think it's wise to:
1. keep the streamTypeLookupWriteOrCreate for future use.
2. don't embed it's alternative code inside xaddCommand, but instead keep it a separate function that has a clear purpose (improves readability) and can be used elsewhere, or even add a `no_create` argument to streamTypeLookupWriteOrCreate.

Reply:
thanks @oranagra , the reason I deleted streamTypeLookupWriteOrCreate is I referenced with other command implementation,  I saw some inconsistencies for different command impementation, for example https://github.com/redis/redis/blob/unstable/src/t_set.c#L268 we embedded the search+create in the command implementation, but https://github.com/redis/redis/blob/unstable/src/t_hash.c#L451 we have a separate function. do you think we need to provide a consistant way for these functions, or just change the stream standalone is fine for now?

Reply:
Consistency is good idea, but i won't go out of my way in this case (i also don't like to mess up the blame log unnecessarily).
In this case what bothered me is readability of the code and possible code sharing in the future.
If the code isn't scattered across multiple areas of the calling function, and has a clear role that can be reused, it's best to name it as a function.
I probably wouldn't notice if it was new code, but it caught my eye when you converted it from a function to embedded code.
So my advise is to just change it in stream (possibly also resulting in a smaller diff)

Reply:
@oranagra thanks Oran, please review my changes, thanks!